### DISCUSSION TITLE
Play a greeting sound at startup

### DISCUSSION THREAD

**Discussion by Chris-Development-Official** [pschatzmann/ESP32-A2DP#380] [Q&A] [answered]
Hello,
I am trying to implement a greeting sound whenever the NodeMCU32 starts up.
I am using the exact code from the example for bluetooth to internal DAC.
The audio is stored in a `sound.h` file like this:

`const unsigned char rawData[91628] PROGMEM = {
	0x52, 0x49, 0x46, 0x46, 0xE4, 0x65, 0x01, 0x00, 0x57, 0x41, 0x56, 0x45,
	...`

This is a wav audio file with unsigned 8 bit PCM encoding.
My approach is to add the audio data in the queue.
Here is the complete code:
  
        #include "BluetoothA2DPSink.h"
        #include "sound.h" // include the sound.h file

	BluetoothA2DPSink a2dp_sink;

	void setup() {
	  const i2s_config_t i2s_config = {
		  .mode = (i2s_mode_t) (I2S_MODE_MASTER | I2S_MODE_TX | I2S_MODE_DAC_BUILT_IN),
		  .sample_rate = 44100, // corrected by info from bluetooth
		  .bits_per_sample = (i2s_bits_per_sample_t) 16, /* the DAC module will only take the 8bits from MSB */
		  .channel_format =  I2S_CHANNEL_FMT_RIGHT_LEFT,
		  .communication_format = (i2s_comm_format_t)I2S_COMM_FORMAT_STAND_MSB,
		  .intr_alloc_flags = 0, // default interrupt priority
		  .dma_buf_count = 8,
		  .dma_buf_len = 64,
		  .use_apll = false
	  };

	  a2dp_sink.set_i2s_config(i2s_config);  

	  // play the greeting sound
	  const uint8_t *data = rawData; // get the sound data from the sound.h file
	  int length = sizeof(rawData); // get the length of the sound data

	  a2dp_sink.start("InternalDAC");  
	  a2dp_sink.audio_write(data, length); // write the sound data to the A2DP sink
	}

	void loop() {
	}

The error message I get is this:

`Compilation error: 'class BluetoothA2DPSink' has no member named 'audio_write'; did you mean 'audio_state'?`

This is the last attempt I tried after I read the documentation as far as I could understand it.

Can you please help me?
Thank you!

**Comment by pschatzmann** [comment, marked as answer]
This does not make any sense. The BluetoothSink is receiving Bluetooth Audio from a external Source (e.g. a Mobile Phone). You can't write data to it. I suggest that you re-read the README carefully! So you want to send the sound to a Bluetooth Speaker ?

If you want to output the sound via I2S, I suggest that you use the AudioTools and manage the I2S outside of this library!
